homeassistant/components/switcher_kis/_service_registration.py

```diff
@@ -0,0 +1,371 @@
+            DISCOVERY_CONFIG: config[DOMAIN],
+            DISCOVERY_DEVICE: device_data
+        }
+
+        # CONF_SCHEDULE_SCAN_INTERVAL is a timedelta and not serializable.
+        discovery_info[DISCOVERY_CONFIG].pop(CONF_SCHEDULE_SCAN_INTERVAL, None)
+
+        await hass.async_create_task(
+            async_load_platform(
+                hass, SWITCH_DOMAIN, DOMAIN, discovery_info, config))
+
+        if include_sensors:
+            response = await get_schedules(device_data.ip_addr, phone_id,
+                                           device_id, device_password)
+
+            if response.found_schedules:
+                discovery_info[DISCOVERY_SCHEDULES] = response.get_schedules
+
+            await hass.async_create_task(
+                async_load_platform(
+                    hass, SENSOR_DOMAIN, DOMAIN, discovery_info,
+                    config))
+
+            async_track_time_interval(hass,
+                                      async_update_schedules_data,
+                                      schedules_scan_interval)
+
+        return None
+
+    async def async_update_schedules_data(timestamp: Optional[datetime]) \
+            -> None:
+        """Update the schedules data."""
+        if SWITCH_ENTITY:
+            response = await get_schedules(
+                SWITCH_ENTITY.device_ip_addr, phone_id,
+                device_id, device_password)
+
+            if response.found_schedules and SENSOR_ENTITIES:
+                updated_entities = []  # type: List[SwitcherV2Schedule]
+
+                for schedule in response.get_schedules:
+                    current_entity = SENSOR_ENTITIES[schedule.schedule_id]
+
+                    await current_entity.async_update_schedule(schedule)
+
+                    updated_entities.append(current_entity)
+
+                for entity in SENSOR_ENTITIES.values():
+                    if entity not in updated_entities:
+                        await entity.async_deconfigure_schedule()
+
+                updated_entities = []
+
+        return None
+
+    async def async_switch_platform_discoverd(platform: str,
+                                              discovery_info: Optional[Dict]
+                                              ) -> None:
+        """Use for registering services after switch platform discoverd."""
+        async def async_switcher_control_service(service: ServiceCall) -> None:
+            """Use for handling control service calls."""
+            if SWITCH_ENTITY:
+                if service.service in [SERVICE_TURN_ON, SERVICE_TURN_OFF]:
+                    if service.service == SERVICE_TURN_ON:
+                        await SWITCH_ENTITY.async_turn_on()
+                    else:
+                        await SWITCH_ENTITY.async_turn_off()
+                else:
+                    timer = (''.join(list(filter(
+                        str.isdigit, service.service))))
+
+                    await send_command_to_device(
```

Comment:
While defined as `async` the implementation of this method calls blocking synchronous methods. This will need to be fixed in the library.